### DISCUSSION TITLE
[Feature request] Possibility to use dependency class name for Body instead of generated name

### DISCUSSION THREAD

**Discussion by AlexanderPodorov** [fastapi/fastapi#10023] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from dataclasses import dataclass
from typing import Annotated

from fastapi import Depends, FastAPI, File, Form, UploadFile
from pydantic import BaseModel, TypeAdapter

api = FastAPI()


@dataclass
class Test:
    a1: Annotated[str, Form()]
    a2: Annotated[str, Form()]
    a3: Annotated[UploadFile, File()]


TestDep = Annotated[Test, Depends()]


fake_items_db = [{"item_name": "Foo"}, {"item_name": "Bar"}, {"item_name": "Baz"}]


class ItemOut(BaseModel):
    item_name: str


@api.get("/items")
async def read_items(t: TestDep) -> ItemOut:
    item = TypeAdapter(ItemOut).validate_python(fake_items_db[0])
    return ItemOut.model_validate(item)
```

### Description

When we use non-JSON body parameters, e.g. `Form` or `File`, we got the request body schema name generated by FastAPI:
![image](https://github.com/tiangolo/fastapi/assets/54511144/fc22fbfa-3bfa-4a4d-93b6-540b36bd3de6)

If we use a user-defined class (`Test` in my example) to wrap **all** body params into it, can FastAPI use this name? It would be ideal. If it's not possible for any reason, can we have a way to override auto-generated name?
I think it would be reasonable to use class name in case if **all** params are wrapped by this class, otherwise auto-generate the name.

### Operating System

macOS

### FastAPI Version

0.101.0

### Pydantic Version

2.1.1

### Python Version

Python 3.11.4

**Comment by YuriiMotov** [comment, marked as answer]
I think this is not possible.

The `Body_read_items_items_get` that you see in openapi is the schema of whole request body of the endpoint.
Following the way you described, if we use this type in several endpoints, the request body schema should be named the same way (`TestDep`?).
But what if second endpoint includes other `Form` parameters along with this dependency? All these parameters should be included into the request body schema.

This is why FastAPI creates different request body schemas for every endpoint and uses the endpoint function name and http-method  to construct the name of this schema.

You can modify the way those name are being created using `generate_unique_id_function` parameter:
```python
counter = {"request_body": 0}

def generate_unique_id(route: APIRoute) -> str:
    counter["request_body"] += 1
    return str(counter["request_body"])

app = FastAPI(generate_unique_id_function=generate_unique_id)
```

By default this one is used: 
https://github.com/fastapi/fastapi/blob/b2d742074c41b4435775e0b24500eb0fe0cb2ed3/fastapi/utils.py#L179-L184
